src/elements/earn/pools/addLiquidity/single/AddLiquiditySingle.tsx

```diff
@@ -32,15 +32,24 @@ export const AddLiquiditySingle = ({ pool }: Props) => {
   const tkn = useAppSelector<Token | undefined>(
     getTokenById(pool.reserves[0].address)
   );
+  const bnt = useAppSelector<Token | undefined>(
+    getTokenById(pool.reserves[1].address)
+  );
   const history = useHistory();
   const approveContract = useRef('');
-  const [selectedToken, setSelectedToken] = useState<Token>(tkn!);
+  const [isBNTSelected, setIsBNTSelected] = useState(false);
   const [amount, setAmount] = useState('');
   const [amountUsd, setAmountUsd] = useState('');
   const [errorMsg, setErrorMsg] = useState('');
   const [spaceAvailableBnt, setSpaceAvailableBnt] = useState('');
   const [spaceAvailableTkn, setSpaceAvailableTkn] = useState('');
 
+  const selectedToken = isBNTSelected ? bnt! : tkn!;
+  const setSelectedToken = (token: Token) => {
```

Comment:
useCallback - or you force child elements to re-render when you re-render. Seems to be the case with other callbacks in this component as well.
Then please go to line 159 and remove the unnecessary arrow function - for the same reason.